configuration.php

```diff
@@ -3,9 +3,19 @@
 /** @var \Icinga\Application\Modules\Module $this */
 
 /** @var \Icinga\Application\Modules\MenuItemContainer $section */
+
+use Icinga\Authentication\Auth;
+use Icinga\Module\Graphite\ProvidedHook\Icingadb\IcingadbSupport;
+
 $section = $this->menuSection(N_('Graphite'), ['icon' => 'chart-area']);
-$section->add(N_('Hosts'), ['url' => 'graphite/list/hosts?graphs_limit=2']);
-$section->add(N_('Services'), ['url' => 'graphite/list/services?graphs_limit=2']);
+
+if ($this::exists('icingadb') && Auth::getInstance()->getUser() !== null && IcingadbSupport::useIcingaDbAsBackend()) {
```

Comment:
I see why you're checking for a user, but I think it's a bad idea to require another check for callers of `useIcingaDbAsBackend()`. The latter should automatically avoid accessing the preferences if there's no logged in user. Please change this in Icinga DB Web instead.